### DISCUSSION TITLE
Cannot pass a defaultValue to the controllable components

### DISCUSSION THREAD

**Discussion by samantagavare** [lookfirst/mui-rff#542] [Q&A]
**Describe the bug**
Restoring values from browser session storage or somewhere else to a controllable Autocomplete/TextField/etc. component is not possible using defaultValue or value parameters. Providing initialValues to the react-final-form Form component also does not work (values are not be visible inside fields). Providing a non-empty value to the component and filled initialValues to the form does work, but it is difficult to maintain these values separately and it is not possible to mark the fields as dirty. In Material-UI, the defaultValue parameter is only used for uncontrollable component, but if the react-final-form Field allows to specify the defaultValue, why not to use it?

**Workaround for Autocomplete component** 
1.  Copy the contents of the [Autocomplete.tsx](https://github.com/lookfirst/mui-rff/blob/master/src/Autocomplete.tsx) file to a new file in your project.
2. Add this:
`defaultValue={rest.defaultValue}`
after
https://github.com/lookfirst/mui-rff/blob/0ba99ae24af40cd810bbe79d31cf2c4062c34009/src/Autocomplete.tsx#L51

**To Reproduce**
1. Open Sandbox link:
- Provided defaultValue to the Autocomplete component: [link](https://codesandbox.io/s/react-final-form-material-ui-example-forked-ksry3?file=/src/index.tsx)
**Result:** Autocomplete field is empty.
**Expected result:** Autocomplete field should have a value of 'Mars' and form field data should have a key "planet" with value [
    "mars"
  ].

**Desktop**
 - OS: Windows
 - Browser: Chrome
 - Version:  93.0.4577.63

**Comment by lookfirst** [comment]
Thanks for your submission. I've converted this to a discussion since I'm not sure this is an issue right now.

I looked at your codesandbox and it isn't clear to me why you aren't using initialValues.planet=['mars']
